### ISSUE TITLE
Docker image build fails on Ubuntu 18.04 and docker 20.10.10

### ISSUE THREAD

**Issue by vk1z** [pytorch/serve#1305]
Docker image build fails in both buildtype production and dev but with different errors - both related to building dependent python packages

Building from master branch, f02a56bf

Failure when building dev image

 Command: cd docker; ./build.sh -bt dev

117.0   Building wheel for pygit2 (pyproject.toml): started
#10 117.6   Building wheel for pygit2 (pyproject.toml): finished with status 'error'
#10 117.6   ERROR: Command errored out with exit status 1:
#10 117.6    command: /usr/bin/python3 /usr/local/lib/python3.6/dist-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmpt88r6n9o
#10 117.6        cwd: /tmp/pip-install-pr_6s8kl/pygit2_0c31181bb64c4dc29b649a8a688b87bb
#10 117.6   Complete output (64 lines):
#10 117.6   running bdist_wheel
#10 117.6   running build
#10 117.6   running build_py
#10 117.6   creating build
#10 117.6   creating build/lib.linux-x86_64-3.6
#10 117.6   creating build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/index.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/repository.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/settings.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/ffi.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/config.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/__init__.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/blame.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/refspec.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/_build.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/_run.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/submodule.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/packbuilder.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/utils.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/errors.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/remote.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/callbacks.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   copying pygit2/credentials.py -> build/lib.linux-x86_64-3.6/pygit2
#10 117.6   creating build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/indexer.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/attr.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/merge.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/types.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/blame.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/pack.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/common.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/clone.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/checkout.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/index.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/net.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/errors.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/oid.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/refspec.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/config.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/submodule.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/remote.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/strarray.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/transport.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/buffer.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/diff.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/revert.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/repository.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/graph.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/callbacks.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/stash.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/proxy.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   copying pygit2/decl/describe.h -> build/lib.linux-x86_64-3.6/pygit2/decl
#10 117.6   running build_ext
#10 117.6   generating cffi module 'build/temp.linux-x86_64-3.6/pygit2._libgit2.c'
#10 117.6   creating build/temp.linux-x86_64-3.6
#10 117.6   building 'pygit2._pygit2' extension
#10 117.6   creating build/temp.linux-x86_64-3.6/src
#10 117.6   x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/local/include -I/usr/include/python3.6m -c src/blob.c -o build/temp.linux-x86_64-3.6/src/blob.o
#10 117.6   In file included from src/blob.c:30:0:
#10 117.6   src/blob.h:33:10: fatal error: git2.h: No such file or directory
#10 117.6    #include <git2.h>
#10 117.6             ^~~~~~~~
#10 117.6   compilation terminated.
#10 117.6   error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
#10 117.6   ----------------------------------------
#10 117.6   ERROR: Failed building wheel for pygit2 
...
 Traceback (most recent call last):
#10 165.3   File "ts_scripts/install_from_src.py", line 70, in <module>
#10 165.3     from pygit2 import Repository
#10 165.3 ModuleNotFoundError: No module named 'pygit2'

executor failed running [/bin/sh -c if [ "$MACHINE_TYPE" = "gpu" ]; then export USE_CUDA=1; fi     && git clone https://github.com/pytorch/serve.git     && cd serve     && git checkout ${BRANCH_NAME}     && if [ -z "$CUDA_VERSION" ]; then python ts_scripts/install_dependencies.py --environment=dev; else python ts_scripts/install_dependencies.py --environment=dev  --cuda $CUDA_VERSION; fi     && python ts_scripts/install_from_src.py     && useradd -m model-server     && mkdir -p /home/model-server/tmp     && cp docker/dockerd-entrypoint.sh /usr/local/bin/dockerd-entrypoint.sh     && chmod +x /usr/local/bin/dockerd-entrypoint.sh     && chown -R model-server /home/model-server     && cp docker/config.properties /home/model-server/config.properties     && mkdir /home/model-server/model-store && chown -R model-server /home/model-server/model-store]: exit code: 1

Production build also fails: cd docker; ./build.sh -bt production

 Running setup.py install for pillow: finished with status 'error'
#13 41.39     Complete output from command /home/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-u0341ra7/pillow/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-cd9c6ujc-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/venv/include/site/python3.6/pillow:
#13 41.39     /usr/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'long_description_content_type'
#13 41.39       warnings.warn(msg)
#13 41.39     running install
#13 41.39     running build
#13 41.39     running build_py
#13 41.39     creating build
#13 41.39     creating build/lib.linux-x86_64-3.6
#13 41.39     creating build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PcfFontFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/Image.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/WmfImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ExifTags.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/WalImageFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/MicImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageTk.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PdfParser.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/FtexImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageMorph.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/XpmImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageCms.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageTransform.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageDraw.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PpmImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/__main__.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageMode.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/FpxImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/SgiImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/TiffTags.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/GimpPaletteFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PixarImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/GimpGradientFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PsdImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImtImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/GbrImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/JpegPresets.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageFilter.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageGrab.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageOps.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/GifImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/GdImageFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageWin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageQt.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/MpegImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/_tkinter_finder.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageMath.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/__init__.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/DcxImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/_util.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/JpegImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PcxImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/XbmImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PyAccess.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/EpsImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageStat.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/TarIO.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/FontFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/MpoImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImagePath.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/DdsImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageDraw2.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/_version.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/WebPImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PdfImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/CurImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/BmpImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ContainerIO.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/TiffImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageColor.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImagePalette.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/SpiderImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/features.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/McIdasImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageChops.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/TgaImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/FliImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PalmImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageFont.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/BufrStubImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/BdfFontFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageShow.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PaletteFile.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageEnhance.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/_binary.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/IcnsImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/BlpImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PSDraw.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/XVThumbImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/GribStubImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/Jpeg2KImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/IcoImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PcdImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/FitsStubImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/SunImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/IptcImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/MspImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/PngImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/Hdf5StubImagePlugin.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     copying src/PIL/ImageSequence.py -> build/lib.linux-x86_64-3.6/PIL
#13 41.39     running egg_info
#13 41.39     writing src/Pillow.egg-info/PKG-INFO
#13 41.39     writing dependency_links to src/Pillow.egg-info/dependency_links.txt
#13 41.39     writing top-level names to src/Pillow.egg-info/top_level.txt
#13 41.39     reading manifest file 'src/Pillow.egg-info/SOURCES.txt'
#13 41.39     reading manifest template 'MANIFEST.in'
#13 41.39     warning: no files found matching '*.c'
#13 41.39     warning: no files found matching '*.h'
#13 41.39     warning: no files found matching '*.sh'
#13 41.39     warning: no previously-included files found matching '.appveyor.yml'
#13 41.39     warning: no previously-included files found matching '.clang-format'
#13 41.39     warning: no previously-included files found matching '.coveragerc'
#13 41.39     warning: no previously-included files found matching '.editorconfig'
#13 41.39     warning: no previously-included files found matching '.readthedocs.yml'
#13 41.39     warning: no previously-included files found matching 'codecov.yml'
#13 41.39     warning: no previously-included files matching '.git*' found anywhere in distribution
#13 41.39     warning: no previously-included files matching '*.pyc' found anywhere in distribution
#13 41.39     warning: no previously-included files matching '*.so' found anywhere in distribution
#13 41.39     no previously-included directories found matching '.ci'
#13 41.39     writing manifest file 'src/Pillow.egg-info/SOURCES.txt'
#13 41.39     running build_ext
#13 41.39     
#13 41.39     
#13 41.39     The headers or library files could not be found for zlib,
#13 41.39     a required dependency when compiling Pillow from source.
#13 41.39     
#13 41.39     Please see the install instructions at:
#13 41.39        https://pillow.readthedocs.io/en/latest/installation.html
#13 41.39     
#13 41.39     Traceback (most recent call last):
#13 41.39       File "/tmp/pip-build-u0341ra7/pillow/setup.py", line 1024, in <module>
#13 41.39         zip_safe=not (debug_build() or PLATFORM_MINGW),
#13 41.39       File "/home/venv/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
#13 41.39         return distutils.core.setup(**attrs)
#13 41.39       File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
#13 41.39         dist.run_commands()
#13 41.39       File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
#13 41.39         self.run_command(cmd)
#13 41.39       File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
#13 41.39         cmd_obj.run()
#13 41.39       File "/home/venv/lib/python3.6/site-packages/setuptools/command/install.py", line 61, in run
#13 41.39         return orig.install.run(self)
#13 41.39       File "/usr/lib/python3.6/distutils/command/install.py", line 589, in run
#13 41.39         self.run_command('build')
#13 41.39       File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
#13 41.39         self.distribution.run_command(command)
#13 41.39       File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
#13 41.39         cmd_obj.run()
#13 41.39       File "/usr/lib/python3.6/distutils/command/build.py", line 135, in run
#13 41.39         self.run_command(cmd_name)
#13 41.39       File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
#13 41.39         self.distribution.run_command(command)
#13 41.39       File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
#13 41.39         cmd_obj.run()
#13 41.39       File "/home/venv/lib/python3.6/site-packages/setuptools/command/build_ext.py", line 78, in run
#13 41.39         _build_ext.run(self)
#13 41.39       File "/usr/lib/python3.6/distutils/command/build_ext.py", line 339, in run
#13 41.39         self.build_extensions()
#13 41.39       File "/tmp/pip-build-u0341ra7/pillow/setup.py", line 790, in build_extensions
#13 41.39         raise RequiredDependencyException(f)
#13 41.39     __main__.RequiredDependencyException: zlib
#13 41.39     
#13 41.39     During handling of the above exception, another exception occurred:
#13 41.39     
#13 41.39     Traceback (most recent call last):
#13 41.39       File "<string>", line 1, in <module>
#13 41.39       File "/tmp/pip-build-u0341ra7/pillow/setup.py", line 1037, in <module>
#13 41.39         raise RequiredDependencyException(msg)
#13 41.39     __main__.RequiredDependencyException:
#13 41.39     
#13 41.39     The headers or library files could not be found for zlib,
#13 41.39     a required dependency when compiling Pillow from source.
#13 41.39     
#13 41.39     Please see the install instructions at:
#13 41.39        https://pillow.readthedocs.io/en/latest/installation.html
#13 41.39     
#13 41.39     
#13 41.39     

The expected behavior in both cases is that images are made.

**Comment by ricdodds** [comment]
Getting the same error when running, either from `master` or `v0.4.2` tag
```
./build_image.sh
...
#13 28.17     During handling of the above exception, another exception occurred:
#13 28.17
#13 28.17     Traceback (most recent call last):
#13 28.17       File "<string>", line 1, in <module>
#13 28.17       File "/tmp/pip-build-ajztltvg/pillow/setup.py", line 1037, in <module>
#13 28.17         raise RequiredDependencyException(msg)
#13 28.17     __main__.RequiredDependencyException:
#13 28.17
#13 28.17     The headers or library files could not be found for zlib,
#13 28.17     a required dependency when compiling Pillow from source.
#13 28.17
#13 28.17     Please see the install instructions at:
#13 28.17        https://pillow.readthedocs.io/en/latest/installation.html
#13 28.17
#13 28.17
#13 28.17
#13 28.17     ----------------------------------------
#13 28.66 Command "/home/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ajztltvg/pillow/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-4lsjh7uh-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/venv/include/site/python3.6/pillow" failed with error code 1 in /tmp/pip-build-ajztltvg/pillow/
------
executor failed running [/bin/sh -c TORCH_VER=$(curl --silent --location https://pypi.org/pypi/torch/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.stdin)['releases']; print(sorted(releases, key=pkg_resources.parse_version)[-1])") &&     TORCH_VISION_VER=$(curl --silent --location https://pypi.org/pypi/torchvision/json | python -c "import sys, json, pkg_resources; releases = json.load(sys.stdin)['releases']; print(sorted(releases, key=pkg_resources.parse_version)[-1])") &&     if echo "$BASE_IMAGE" | grep -q "cuda:"; then         if [ "$CUDA_VERSION" ]; then             pip install --no-cache-dir torch==$TORCH_VER+$CUDA_VERSION torchvision==$TORCH_VISION_VER+$CUDA_VERSION -f https://download.pytorch.org/whl/torch_stable.html;         else             pip install --no-cache-dir torch torchvision;         fi     else         pip install --no-cache-dir torch==$TORCH_VER+cpu torchvision==$TORCH_VISION_VER+cpu -f https://download.pytorch.org/whl/torch_stable.html;     fi]: exit code: 1
```

**Comment by vk1z** [comment]
Is there any good reason to build pillow from sources rather than simply installing it via pip?

**Comment by elk-cloner** [comment]
> Is there any good reason to build pillow from sources rather than simply installing it via pip?

pillow depends on some libs (`zlib1g-dev`,  `libjpeg8-dev`) that aren't in docker file, you can add them [here](https://github.com/pytorch/serve/blob/72761c050b4c05616a02c917844e1492c9dc8611/docker/Dockerfile#L33).

**Comment by aaronzs** [comment]
Specify pillow version to Pillow==8.2.0 will install pillow using wheel rather than building from source.

**Comment by lxning** [comment]
The root cause is pitgit2 1.7.0 has issue with link.
solution: lock pygit2==1.6.1 in requirements/developer.txt.

It is fixed in v0.4.3.
